### ISSUE TITLE
Thread 'libp2p-0' panicked at 'assertion failed: listener.addresses.contains(&listen_addr)'

### ISSUE THREAD

**Issue by starit** [paritytech/substrate#3352]
I am running a node on server with dev mode and it shows the following error:

```
2019-08-08 14:21:10 Prepared block for proposing at 79749 [hash: 0x2966cd0610084262f28e3f12c813355a871aebd058dce67d9d94102739dccb72; parent_hash: 0x669a…75ca; extrinsics: [0xfe50…384e]]
2019-08-08 14:21:10 Pre-sealed block for proposal at 79749. Hash now 0xa110b6e54eab2bdcabd1a1071347b655996b409c4fd2bcbc2b30a1769dfae589, previously 0x2966cd0610084262f28e3f12c813355a871aebd058dce67d9d94102739dccb72.
2019-08-08 14:21:10 Imported #79749 (0xa110…e589)
2019-08-08 14:21:14 Idle (0 peers), best: #79749 (0xa110…e589), finalized #0 (0x9f99…18b6), ⬇ 0.4kiB/s ⬆ 0.5kiB/s
2019-08-08 14:21:16 Libp2p => Random Kademlia query has yielded empty results
2019-08-08 14:21:19 Idle (0 peers), best: #79749 (0xa110…e589), finalized #0 (0x9f99…18b6), ⬇ 37 B/s ⬆ 89 B/s
2019-08-08 14:21:20 Starting consensus session on top of parent 0xa110b6e54eab2bdcabd1a1071347b655996b409c4fd2bcbc2b30a1769dfae589
2019-08-08 14:21:20 Prepared block for proposing at 79750 [hash: 0xc371f005bd898603a28129f8000c342b09750702264a1fd22c8bb418fe68d375; parent_hash: 0xa110…e589; extrinsics: [0xe3eb…86e7]]
2019-08-08 14:21:20 Pre-sealed block for proposal at 79750. Hash now 0x050423fc6d5ba985a18844a86c0110a3aabb657103aaaff8c4edfcdbf48b0a01, previously 0xc371f005bd898603a28129f8000c342b09750702264a1fd22c8bb418fe68d375.
2019-08-08 14:21:20 Imported #79750 (0x0504…0a01)
2019-08-08 14:21:24 Idle (0 peers), best: #79750 (0x0504…0a01), finalized #0 (0x9f99…18b6), ⬇ 0.9kiB/s ⬆ 1.2kiB/s

====================

stack backtrace:
   0: substrate_panic_handler::panic_hook::hf42adcb88622a58f (0x55f3a28b42e4)
             at /home/jamie/.cargo/git/checkouts/substrate-7e08433d4c370a21/2e30b73/core/panic-handler/src/lib.rs:82
   1: substrate_panic_handler::set::{{closure}}::hb8e3e22d25ad84e5 (0x55f3a28b3e87)
             at /home/jamie/.cargo/git/checkouts/substrate-7e08433d4c370a21/2e30b73/core/panic-handler/src/lib.rs:31
   2: std::panicking::rust_panic_with_hook::h94f6d90998bef5c8 (0x55f3a2df10d9)
             at src/libstd/panicking.rs:481
   3: std::panicking::begin_panic::h2500a58a097b3e83 (0x55f3a263abc8)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/panicking.rs:411
   4: libp2p_core::nodes::listeners::ListenersStream<TTrans>::poll::hd75cf4e84411eb2e (0x55f3a0253d1a)
             at /home/jamie/substrate/definex-poc-1/<::std::macros::panic macros>:4
   5: libp2p_core::nodes::raw_swarm::RawSwarm<TTrans,TInEvent,TOutEvent,THandler,THandlerErr,TConnInfo,TPeerId>::poll::h8b15eb0831b45e2e (0x55f3a0e76a3f)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/libp2p-core-0.7.1/src/nodes/raw_swarm.rs:1003
   6: <libp2p_core::swarm::swarm::ExpandedSwarm<TTransport,TBehaviour,TInEvent,TOutEvent,THandler,THandlerErr> as futures::stream::Stream>::poll::hb3fdc936d916e92e (0x55f3a06a69d8)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/libp2p-core-0.7.1/src/swarm/swarm.rs:275
   7: substrate_network_libp2p::service_task::Service<TMessage>::poll_swarm::hd025724873f58c0e (0x55f3a1145130)
             at /home/jamie/.cargo/git/checkouts/substrate-7e08433d4c370a21/2e30b73/core/network-libp2p/src/service_task.rs:312
   8: <substrate_network_libp2p::service_task::Service<TMessage> as futures::stream::Stream>::poll::hb752aacdfc55fd52 (0x55f3a1142563)
             at /home/jamie/.cargo/git/checkouts/substrate-7e08433d4c370a21/2e30b73/core/network-libp2p/src/service_task.rs:379
   9: substrate_network::service::run_thread::{{closure}}::hcaf26a139eadf48f (0x55f3a0e41325)
             at /home/jamie/.cargo/git/checkouts/substrate-7e08433d4c370a21/2e30b73/core/network/src/service.rs:551
  10: <futures::stream::poll_fn::PollFn<F> as futures::stream::Stream>::poll::h2b41101dc963e71f (0x55f3a10b0ada)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/stream/poll_fn.rs:47
  11: <futures::stream::for_each::ForEach<S,F,U> as futures::future::Future>::poll::h4c4ffba08949b559 (0x55f3a071c051)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/stream/for_each.rs:45
  12: <alloc::boxed::Box<F> as futures::future::Future>::poll::h4fb3d43e36c78b39 (0x55f3a030c15c)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/future/mod.rs:113
  13: <futures::future::select_all::SelectAll<A> as futures::future::Future>::poll::{{closure}}::he9e6f400e9108e94 (0x55f3a072e469)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/future/select_all.rs:53
  14: <core::iter::adapters::FilterMap<I,F> as core::iter::traits::iterator::Iterator>::try_fold::{{closure}}::h8b50ce43521737d9 (0x55f3a049fe9e)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libcore/iter/adapters/mod.rs:826
  15: <core::iter::adapters::Enumerate<I> as core::iter::traits::iterator::Iterator>::try_fold::{{closure}}::h1d0d88a89260a734 (0x55f3a04ce995)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libcore/iter/adapters/mod.rs:951
  16: <core::slice::IterMut<T> as core::iter::traits::iterator::Iterator>::try_fold::ha2fe4434b160fdcf (0x55f3a0498748)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libcore/slice/mod.rs:3164
  17: <core::iter::adapters::Enumerate<I> as core::iter::traits::iterator::Iterator>::try_fold::h0c09072b1af57538 (0x55f3a04ce659)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libcore/iter/adapters/mod.rs:950
  18: <core::iter::adapters::FilterMap<I,F> as core::iter::traits::iterator::Iterator>::try_fold::haaa12c38560433e4 (0x55f3a049f194)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libcore/iter/adapters/mod.rs:826
  19: core::iter::traits::iterator::Iterator::try_for_each::h5cabc3efa6792cfb (0x55f3a04a34ba)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libcore/iter/traits/iterator.rs:1610
  20: <core::iter::adapters::FilterMap<I,F> as core::iter::traits::iterator::Iterator>::next::h2a8a49fe1dc8c811 (0x55f3a049ebc8)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libcore/iter/adapters/mod.rs:812
  21: <futures::future::select_all::SelectAll<A> as futures::future::Future>::poll::h1c196370e6599ad5 (0x55f3a072df41)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/future/select_all.rs:52
  22: futures::future::chain::Chain<A,B,C>::poll::hf426b6016685ad8f (0x55f3a0864e1d)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/future/chain.rs:26
  23: <futures::future::and_then::AndThen<A,B,F> as futures::future::Future>::poll::h6b4053e6378a9558 (0x55f3a113737a)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/future/and_then.rs:32
  24: <futures::future::map_err::MapErr<A,F> as futures::future::Future>::poll::h1fcd5da92eaac377 (0x55f3a04d35d3)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/future/map_err.rs:30
  25: <futures::future::select::Select<A,B> as futures::future::Future>::poll::he5df7bc5bb3ed14f (0x55f3a03cf1ce)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/future/select.rs:48
  26: <futures::future::map::Map<A,F> as futures::future::Future>::poll::h700d14369dbe8b26 (0x55f3a0dc4b7e)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/future/map.rs:30
  27: <futures::future::map_err::MapErr<A,F> as futures::future::Future>::poll::h3f2509ce4c0906c5 (0x55f3a04d4756)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/future/map_err.rs:30
  28: futures::future::chain::Chain<A,B,C>::poll::hd5f93d7bfb9f9547 (0x55f3a086280a)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/future/chain.rs:26
  29: <futures::future::then::Then<A,B,F> as futures::future::Future>::poll::h945753ca9c24302e (0x55f3a03be8a3)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/future/then.rs:32
  30: <alloc::boxed::Box<F> as futures::future::Future>::poll::h66f74293210c7ab2 (0x55f3a27e9735)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/future/mod.rs:113
  31: futures::task_impl::Spawn<T>::poll_future_notify::{{closure}}::h88fd63558feea88d (0x55f3a2729094)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/task_impl/mod.rs:329
  32: futures::task_impl::Spawn<T>::enter::{{closure}}::h1ea0dda1a985d830 (0x55f3a2729341)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/task_impl/mod.rs:399
  33: futures::task_impl::std::set::he3be8b12a125c2df (0x55f3a2720764)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/task_impl/std/mod.rs:83
  34: futures::task_impl::Spawn<T>::enter::h43bbc2886a199663 (0x55f3a27291bf)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/task_impl/mod.rs:399
  35: futures::task_impl::Spawn<T>::poll_fn_notify::hd2fc24147f4fbfab (0x55f3a2728f28)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/task_impl/mod.rs:291
  36: futures::task_impl::Spawn<T>::poll_future_notify::hfdba73dde95ff6ba (0x55f3a2729068)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.28/src/task_impl/mod.rs:329
  37: tokio_threadpool::task::Task::run::{{closure}}::hc957ed4ff6da4aa0 (0x55f3a272ae72)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.15/src/task/mod.rs:145
  38: core::ops::function::FnOnce::call_once::he7bfae3755cea793 (0x55f3a273da09)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libcore/ops/function.rs:231
  39: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h92b6a0c822d4aca2 (0x55f3a272b7f9)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/panic.rs:309
  40: std::panicking::try::do_call::hb2f5de8c45ee82dd (0x55f3a274e233)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/panicking.rs:296
  41: __rust_maybe_catch_panic (0x55f3a2df637a)
             at src/libpanic_unwind/lib.rs:82
  42: std::panicking::try::h3d74ac5382caa04c (0x55f3a274de20)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/panicking.rs:275
  43: std::panic::catch_unwind::ha67dd64377aa5be3 (0x55f3a272c0b1)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/panic.rs:388
  44: tokio_threadpool::task::Task::run::h46c55ab905c7ac33 (0x55f3a272a6a0)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.15/src/task/mod.rs:130
  45: tokio_threadpool::worker::Worker::run_task2::hbc16508a0ffe62f1 (0x55f3a273acdf)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.15/src/worker/mod.rs:567
  46: tokio_threadpool::worker::Worker::run_task::h8caaa89ad911d921 (0x55f3a273a544)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.15/src/worker/mod.rs:459
  47: tokio_threadpool::worker::Worker::try_run_owned_task::hb61161d9c80f8c5c (0x55f3a2739b9d)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.15/src/worker/mod.rs:390
  48: tokio_threadpool::worker::Worker::try_run_task::hb4f10b063a06be93 (0x55f3a27395cd)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.15/src/worker/mod.rs:297
  49: tokio_threadpool::worker::Worker::run::h1029d7e179c0e2f5 (0x55f3a273940b)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.15/src/worker/mod.rs:241
  50: tokio::runtime::threadpool::builder::Builder::build::{{closure}}::{{closure}}::{{closure}}::{{closure}}::h5e363cc3a06e6626 (0x55f3a26c3f2c)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.22/src/runtime/threadpool/builder.rs:390
  51: tokio_timer::timer::handle::with_default::{{closure}}::h1cd6f876c606d257 (0x55f3a26d106c)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.11/src/timer/handle.rs:101
  52: std::thread::local::LocalKey<T>::try_with::h0bff93a5a8deb7d6 (0x55f3a26ccaff)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/thread/local.rs:257
  53: std::thread::local::LocalKey<T>::with::he01892c41376a75b (0x55f3a26cc8f3)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/thread/local.rs:234
  54: tokio_timer::timer::handle::with_default::h3774e81351d55fcf (0x55f3a26d0eb3)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.11/src/timer/handle.rs:84
  55: tokio::runtime::threadpool::builder::Builder::build::{{closure}}::{{closure}}::{{closure}}::h52552e18d4b1fc54 (0x55f3a26c3f8c)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.22/src/runtime/threadpool/builder.rs:382
  56: tokio_timer::clock::clock::with_default::{{closure}}::h9ed6cfcfbeb62cb4 (0x55f3a26bf9bf)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.11/src/clock/clock.rs:137
  57: std::thread::local::LocalKey<T>::try_with::hf221fd01b7e82b94 (0x55f3a26ccf7b)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/thread/local.rs:257
  58: std::thread::local::LocalKey<T>::with::hf119f65df9c5c006 (0x55f3a26cc97d)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/thread/local.rs:234
  59: tokio_timer::clock::clock::with_default::hcfd54c49008e7151 (0x55f3a26bf8bb)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.11/src/clock/clock.rs:117
  60: tokio::runtime::threadpool::builder::Builder::build::{{closure}}::{{closure}}::hf65a776ace7cd0d7 (0x55f3a26c3fdd)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.22/src/runtime/threadpool/builder.rs:381
  61: tokio_reactor::with_default::{{closure}}::ha802c0d7ba64ee62 (0x55f3a26d13ed)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-reactor-0.1.9/src/lib.rs:237
  62: std::thread::local::LocalKey<T>::try_with::h2abfd26781b437bc (0x55f3a26ccc8b)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/thread/local.rs:257
  63: std::thread::local::LocalKey<T>::with::hae6fd888179c4f14 (0x55f3a26cc87d)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/thread/local.rs:234
  64: tokio_reactor::with_default::h7e63c723e5378717 (0x55f3a26d11b5)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-reactor-0.1.9/src/lib.rs:217
  65: tokio::runtime::threadpool::builder::Builder::build::{{closure}}::h0482834b56332030 (0x55f3a26c4080)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.22/src/runtime/threadpool/builder.rs:380
  66: tokio_threadpool::callback::Callback::call::h9bd6574d17443b99 (0x55f3a272fd83)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.15/src/callback.rs:22
  67: tokio_threadpool::worker::Worker::do_run::{{closure}}::{{closure}}::hca668c7867268a83 (0x55f3a27390bf)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.15/src/worker/mod.rs:127
  68: tokio_executor::global::with_default::{{closure}}::h56da1a73f22c7249 (0x55f3a274a0fb)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-executor-0.1.8/src/global.rs:209
  69: std::thread::local::LocalKey<T>::try_with::he6c18abde75f978b (0x55f3a273229f)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/thread/local.rs:257
  70: std::thread::local::LocalKey<T>::with::hbbd6e97307db7632 (0x55f3a27319c3)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/thread/local.rs:234
  71: tokio_executor::global::with_default::hf6573be03bf79b33 (0x55f3a2749fd3)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-executor-0.1.8/src/global.rs:178
  72: tokio_threadpool::worker::Worker::do_run::{{closure}}::ha5e1576d31d8a863 (0x55f3a273916d)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.15/src/worker/mod.rs:125
  73: std::thread::local::LocalKey<T>::try_with::h602be0c22333b33e (0x55f3a2731cb6)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/thread/local.rs:257
  74: std::thread::local::LocalKey<T>::with::ha6b1eca555e3f9bd (0x55f3a273194e)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/thread/local.rs:234
  75: tokio_threadpool::worker::Worker::do_run::h250c0d1047840bbc (0x55f3a2738f95)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.15/src/worker/mod.rs:116
  76: tokio_threadpool::pool::Pool::spawn_thread::{{closure}}::he66d7dd4257d4c49 (0x55f3a27451e0)
             at /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.15/src/pool/mod.rs:344
  77: std::sys_common::backtrace::__rust_begin_short_backtrace::h18c375530b93e8f1 (0x55f3a274da4e)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/sys_common/backtrace.rs:77
  78: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}::hc29cf674cf3e226f (0x55f3a27355d2)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/thread/mod.rs:470
  79: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h221dd5c13993aca0 (0x55f3a272b7c2)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/panic.rs:309
  80: std::panicking::try::do_call::h16d3a0509c5e0a34 (0x55f3a274e1bc)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/panicking.rs:296
  81: __rust_maybe_catch_panic (0x55f3a2df637a)
             at src/libpanic_unwind/lib.rs:82
  82: std::panicking::try::he3c59b91c22a9560 (0x55f3a274e002)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/panicking.rs:275
  83: std::panic::catch_unwind::ha63b8079915d8221 (0x55f3a272c062)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libstd/panic.rs:388
  System information as of Thu Aug  8 14:03:34 UTC 2019

             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/libcore/ops/function.rs:231
  86: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::hfb7da7c10e37eb96 (0x55f3a2ddf79f)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/liballoc/boxed.rs:746
  87: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::hfa4f87915138ff10 (0x55f3a2df5700)
             at /rustc/0beb2ba16a08dfa01569b5f4644da315dc4c806c/src/liballoc/boxed.rs:746
      std::sys_common::thread::start_thread::h15a2ed152517f722
             at src/libstd/sys_common/thread.rs:13
      std::sys::unix::thread::Thread::new::thread_start::h006cc43581e90d8f
             at src/libstd/sys/unix/thread.rs:79
  88: start_thread (0x7f36a8f186db)
  89: __clone (0x7f36a8a2988f)
  90: <unknown> (0x0)

Thread 'libp2p-0' panicked at 'assertion failed: listener.addresses.contains(&listen_addr)', /home/jamie/.cargo/registry/src/github.com-1ecc6299db9ec823/libp2p-core-0.7.1/src/nodes/listeners.rs:191

This is a bug. Please report it at:

        support.anonymous.an
```

**Comment by tomaka** [comment]
cc @twittner 

Let's: 
- Replace that debug_assert! with a `warn!`.
- Print out `listener.addresses` and `listen_addr`.

**Comment by tomaka** [comment]
Should be fixed by #3408
